protocol/storage/client.go

```diff
@@ -340,17 +357,25 @@ type ClientNodeImpl struct {
 	dserv     ipld.DAGService
 	host      host.Host
 	blockTime time.Duration
+	signer    types.Signer
 }
 
 // NewClientNodeImpl constructs a ClientNodeImpl
-func NewClientNodeImpl(ds ipld.DAGService, host host.Host, bt time.Duration) *ClientNodeImpl {
+func NewClientNodeImpl(ds ipld.DAGService, host host.Host, bt time.Duration, s types.Signer) *ClientNodeImpl {
 	return &ClientNodeImpl{
 		dserv:     ds,
 		host:      host,
 		blockTime: bt,
+		signer:    s,
 	}
 }
 
+// SignBytes implements the types.signer interface. This allows ClientNode to
+// sign data with the address `addr`.
+func (cni *ClientNodeImpl) SignBytes(data []byte, addr address.Address) (types.Signature, error) {
```

Comment:
BLOCKING: SignBytes is already available from the plumbing API. https://github.com/filecoin-project/go-filecoin/blob/9549501fa9513cc9141426c7e958a2138ff97890/plumbing/api.go#L177 Since the client already has access to the API, there's no need to reimplement it here.